Chatkit/Redux/State/Types/ListState.swift

```diff
@@ -0,0 +1,40 @@
+
+protocol ListState: State, Sequence, IteratorProtocol where Element == StateElement {
+    
+    associatedtype StateElement: State
+    
+    var elements: [String : StateElement] { get }
+    var iteratorIndexes: [String] { get }
```

Comment:
That is true, there is one problem however. I tried to look this up in the documentation when I initially implemented this and it turns out that Apple does not guarantee the order of `keys` array. Even though this works at the moment, many people on https://forums.swift.org recommend not to relay on the provided order.

Reply:
Sorry can you help me fully understand the issue you've mentioned? 

Dictionaries aren't sorted to begin with so I would expect `keys` to return the keys in an arbitrary order, however if the elements in the dictionary haven't changed then I would expect the order to remain consistent with every call. 

Is the suggestion that on one pass of `next`, `elements.keys` might return `["A", "C", "B"]` (for example) and then on the next pass it might return `[""C", A", "B"]`?

Reply:
As you mentioned, dictionaries aren't sorted. `keys` property returns an array of all keys in an arbitrary order. In practice this order does not change between different calls to `keys`, however that is not guaranteed by Apple, so theoretically this behaviour could change in future (that's probably unlikely to happen). That is why I introduced an additional property in the protocol.

Reply:
OK, seems a little paranoid to me but I'm happy for us to leave it as is.  

I think we will be adding some more `ListState`s in future so at some point it may make sense to move this commonality into a subclass or composite or something.  Anyway TBD.